crates/node/core/src/args/pruning.rs

```diff
@@ -1,50 +1,249 @@
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn receipts_prune_mode(&self) -> Option<PruneMode> {
+        if self.receipts_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.receipts_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.receipts_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn account_history_prune_mode(&self) -> Option<PruneMode> {
+        if self.account_history_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.account_history_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.account_history_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn storage_history_prune_mode(&self) -> Option<PruneMode> {
+        if self.storage_history_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.storage_history_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.storage_history_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    fn parse_receipts_log_filter(&self) -> Result<ReceiptsLogPruneConfig, ReceiptsLogError> {
+        let mut config = BTreeMap::new();
+        // Each filter is provided in the form <address>:<prunemode>
+        for filter in &self.receipts_log_filter {
+            let parts: Vec<&str> = filter.split(':').collect();
+            if parts.len() < 2 {
+                return Err(ReceiptsLogError::InvalidFilterFormat(filter.clone()));
+            }
+            // Parse the address
+            let address = parts[0]
+                .parse::<Address>()
+                .map_err(|_| ReceiptsLogError::InvalidAddress(parts[0].to_string()))?;
+
+            // Parse the prune mode
+            let prune_mode = match parts[1] {
+                "full" => PruneMode::Full,
+                s if s.starts_with("distance") => {
+                    let distance =
+                        parts[2].parse::<u64>().map_err(ReceiptsLogError::InvalidDistance)?;
+                    PruneMode::Distance(distance)
+                }
+                s if s.starts_with("before") => {
+                    let block_number = parts[2]
+                        .parse::<BlockNumber>()
+                        .map_err(ReceiptsLogError::InvalidBlockNumber)?;
+                    PruneMode::Before(block_number)
+                }
+                _ => return Err(ReceiptsLogError::InvalidPruneMode(parts[0].to_string())),
+            };
+            config.insert(address, prune_mode);
+        }
+        Ok(ReceiptsLogPruneConfig(config))
     }
 }
```

Comment:
if this instead took `Vec<String>` as an argument, then it could be moved out into a standalone function instead of be a method of `PruningArgs`, and you could use it in clap attribute like 
https://github.com/paradigmxyz/reth/blob/d5906f3f89653059e37afff71a9913a9b43a5039/crates/cli/commands/src/db/get.rs#L45-L46

Reply:
Thanks for the suggestion